@here/harp-mapview/lib/text/Placement.ts

```diff
@@ -286,8 +307,121 @@ export function placeIcon(
     return PlacementResult.Ok;
 }
 
+/**
+ * Try to place a point label text using multiple optional placements.
+ *
+ * @param labelState State of the point label to place.
+ * @param screenPosition Position of the label in screen coordinates.
+ * @param scale Scale factor to be applied to label dimensions.
+ * @param isRejected Whether the label is already rejected (e.g. because its icon was rejected). If
+ * `true`, text won't be checked for collision, result will be either `PlacementResult.Invisible` or
+ * `PlacementResult.Rejected`.
+ * @param textCanvas The text canvas where the label will be placed.
+ * @param screenCollisions Used to check collisions with other labels.
+ * @param outScreenPosition The final label screen position after applying any offsets.
+ * @returns `PlacementResult.Ok` if path label can be placed at the base or optional anchor point,
+ * `PlacementResult.Rejected` if there's a collision for all placements, `PlacementResult.Invisible`
+ * if it's not visible any any placement position.
+ */
+export function placePointLabelChoosingAnchor(
+    labelState: TextElementState,
+    screenPosition: THREE.Vector2,
+    scale: number,
+    textCanvas: TextCanvas,
+    screenCollisions: ScreenCollisions,
+    isRejected: boolean,
+    outScreenPosition: THREE.Vector3
+): PlacementResult {
+    const label = labelState.element;
+
+    // NOTE:
+    // We may save some CPU time by firstly checking visibility of all possible text placements,
+    // their bounding boxes merged altogether, this may save time for further calculations.
+    // Would be enough to include TL and BR placements.
+
+    // Start with currently set alignment settings.
+    const basePlacement = {
+        h: label.layoutStyle!.horizontalAlignment,
+        v: label.layoutStyle!.verticalAlignment
+    };
+    // Placements options will be read from label.layoutStyle.placements in final solution.
+    const placements = possibleAnchorPlacementsCW;
+    // Find first anchor placement.
+    const idx = placements.findIndex(val => {
+        return val.h === basePlacement.h && val.v === basePlacement.v;
+    });
+    // If not listed in placement options list (idx === -1) will start from current alignment setup.
+    const placementsNum = placements.length;
+    // Will be true if all text placements are invisible.
+    let allInvisible: boolean = true;
+    let i = 0;
+    for (i = 0; i < placementsNum; ++i) {
+        const anchorPlacement = idx + i < 0 ? basePlacement : placements[(idx + i) % placementsNum];
```

Comment:
This loop isn't doing what's expected in several cases:
1) For centered labels we don't want to test other placement.
2) If the original placement is not in the placement options (idx ===-1) then you're only testing placementsNum-1 of the alternative placements (you will never test the last placement in the array, i.e. {Left, Below})!.
3) Since we're mising 4 placements in the array ({Top,Center}, {Bottom,Center}...), if the original placement is one of those, we won't start with the closest alternative placement, but always with {Left, Above}.

Reply:
>1. For centered labels we don't want to test other placement.

And it will never do, but it is a part of different (and temporary logic) - see caller function.

> 2. If the original placement is not in the placement options (idx ===-1) then you're only testing placementsNum-1 of the alternative placements (you will never test the last placement in the array, i.e. {Left, Below})!.

Not true, it is exactly opposite, if original placement is in the options list then we test `placementsNum` iterations, if original is not on the list then we test `placementsNum + 1`, it is desired, and I don't want to add any artificial placement variant the the first scenario.

> 3. Since we're mising 4 placements in the array ({Top,Center}, {Bottom,Center}...), if the original placement is one of those, we won't start with the closest alternative placement, but always with {Left, Above}.

Good point for improvement, I have already changed that in the last update.